### ISSUE TITLE
`assert_type(1, int)` fails

### ISSUE THREAD

**Issue by exoriente** [python/mypy#22031]
**Bug Report**

Using `typing.assert_type` to assert that a integer literal like `1` is of type `int` causes `mypy` to raise an error.

Obviously a check like this is not a very useful application of `assert_type`, but I stumbled across this when trying to figure out a nice way to import `assert_type` after facing issues mentioned here: #14220 (otherwise unrelated to the bug I'm reporting).

**To Reproduce**

```python
from typing import assert_type

assert_type(1, int)
```

**Expected Behavior**

```
Success: no issues found in 1 source file
```

**Actual Behavior**

```
main.py:3: error: Expression is of type "Literal[1]", not "int"  [assert-type]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: `2.3.1`
- Mypy command-line flags: `None`
- Mypy configuration options from `mypy.ini` (and other config files): `None`
- Python version used: `3.14`

**Comment by JelleZijlstra** [comment]
Mypy is correct here: it infers `1` to be of type `Literal[1]`, which is a different type than `int`. To reliably get an exact type inferred, use a function parameter:

```
def f(x: int):
    assert_type(x, int)
```

**Comment by exoriente** [comment]
Clear. I assumed that `reveal_type` functions like `isinstance` except at type-checking time and should pass in this case since `Literal[1]` is a subtype of `int`. I see I was mistaken.
